database/migrations/2022_11_13_212420_create_rooms_table.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('rooms', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('floor_id');
+            $table->string('name');
+            $table->enum('type', ['STANDARD', 'DUPLEX', 'DELUX']);
```

Comment:
I think we should avoid creating DB enums entries like this.
We've discussed this before, enums declared in the DB level are pretty hard and harmful to modify in the future.
Instead, we can do this `$table->string('type'); // this will be a server-side enum attribute`
We can keep the comment so we will understand in the future what's going on here.

Reference: [Youtube video](https://youtu.be/GwunZ5sH2P8?t=1660)

Reply:
I think we won't need to create separate files for defining enums. Try for yourself though